src/core/ext/xds/xds_route_config.cc

```diff
@@ -149,58 +149,53 @@ XdsRouteConfigResource::Route::RouteAction::HashPolicy::operator=(
   return *this;
 }
 
-XdsRouteConfigResource::Route::RouteAction::HashPolicy::HashPolicy(
-    HashPolicy&& other) noexcept
-    : type(other.type),
-      header_name(std::move(other.header_name)),
+XdsRouteConfigResource::Route::RouteAction::HashPolicy::Header::Header(
+    Header&& other) noexcept
+    : header_name(std::move(other.header_name)),
       regex(std::move(other.regex)),
       regex_substitution(std::move(other.regex_substitution)) {}
 
-XdsRouteConfigResource::Route::RouteAction::HashPolicy&
-XdsRouteConfigResource::Route::RouteAction::HashPolicy::operator=(
-    HashPolicy&& other) noexcept {
-  type = other.type;
+XdsRouteConfigResource::Route::RouteAction::HashPolicy::Header&
+XdsRouteConfigResource::Route::RouteAction::HashPolicy::Header::operator=(
+    Header&& other) noexcept {
   header_name = std::move(other.header_name);
   regex = std::move(other.regex);
   regex_substitution = std::move(other.regex_substitution);
   return *this;
 }
 
-bool XdsRouteConfigResource::Route::RouteAction::HashPolicy::HashPolicy::
-operator==(const HashPolicy& other) const {
-  if (type != other.type) return false;
-  if (type == Type::HEADER) {
-    if (regex == nullptr) {
-      if (other.regex != nullptr) return false;
-    } else {
-      if (other.regex == nullptr) return false;
-      return header_name == other.header_name &&
-             regex->pattern() == other.regex->pattern() &&
-             regex_substitution == other.regex_substitution;
-    }
+bool XdsRouteConfigResource::Route::RouteAction::HashPolicy::Header::operator==(
+    const Header& other) const {
+  if (header_name != other.header_name) return false;
```

Comment:
Let's maintain the same order as the declaration of the members - header_name, regex, regex_substitution even when comparing